pep-0596.rst

```diff
@@ -0,0 +1,191 @@
+PEP: 596
+Title: Use UTF-8 for default text file encoding
+Author: Inada Naoki  <songofacandy@gmail.com>
+Status: Draft
+Type: Standards Track
+Content-Type: text/x-rst
+Created: 04-Jun-2019
+Python-Version: 3.9
+
+Abstract
+========
+
+Currently, ``TextIOWrapper`` uses ``locale.getpreferredencoding(False)``
+when ``encoding`` is not specified.
+
+This PEP proposes changing the default encoding to "UTF-8" regardless of
+platform or locale.
+
+
+Motivation
+==========
+
+People assume UTF-8 always
+--------------------------
+
+Package authors using macOS or Linux may forget that the default encoding
+is not always UTF-8.
+
+For example, ``long_description = open("README.md").read()`` in
+``setup.py`` is common mistake.  If there are at least one emoji or any
+other non-ASCII characters in the ``README.md`` file, many Windows users
+cannot install the package by ``UnicodeDecodeError``.
```

Comment:
IMHO we need a way to opt-in for the "locale encoding": current Python 3.8 default encoding when open() if called without specifying an encoding.

I made an attempt to add an "locale" encoding which is the locale encoding.

Reply:
Currently the “locale” encoding (platform encoding?) is available via `open(..., encoding=None)`. Would it be a good idea to keep it available? I.e. instead of changing the behaviour of `encoding=None` to use UTF-8, change the *signature* of `open` to `encoding='utf8'`.

Reply:
> IMHO we need a way to opt-in for the "locale encoding": current Python 3.8 default encoding when open() if called without specifying an encoding.
> 
> I made an attempt to add an "locale" encoding which is the locale encoding.

On Windows, we have "mbcs" already.
How many scripts which works on other than Windows need shortcut for `locale.getpreferredencoding`?


> Currently the “locale” encoding (platform encoding?) is available via `open(..., encoding=None)`. Would it be a good idea to keep it available? I.e. instead of changing the behaviour of `encoding=None` to use UTF-8, change the _signature_ of `open` to `encoding='utf8'`.

I don't think so.  There are many functions which takes `encoding=None` parameter
and pass it to `open(..., encoding=encoding)`.  We can change all caller in stdlib, but
we can not change all callers in other libs.